put/request.go

```diff
@@ -622,3 +628,59 @@ func handleNamespace(key string) (string, error) {
 		return "", errInvalidMetaNamespace
 	}
 }
+
+// ValidateAndCreateReasonMetadata returns an error if the provided string is
+// not a valid ibackup reason, otherwise returns the related reason metadata
+// key.
+func ValidateAndCreateReasonMetadata(reason string) (string, error) {
+	validReasons := []string{"backup", "archive", "quarantine"}
+	if !slices.Contains(validReasons, reason) {
+		return "", errInvalidReasonMeta
+	}
+
+	return MetaKeyReason, nil
+}
+
+// ValidateAndCreateReviewMetadata checks the given duration string is valid and
+// returns the date corresponding to the duration along with the related review
+// metadata key.
+func ValidateAndCreateReviewMetadata(durationTilReview string) (string, time.Time, error) {
+	date, err := getFutureDateFromDuration(durationTilReview)
+
+	return MetaKeyReview, date, err
+}
+
+// getFutureDateFromDuration calculates the future date based on the duration
+// string provided. Returns an error if duration is not in the format
+// '<number><unit>', e.g. '1y', '12m'.
+func getFutureDateFromDuration(duration string) (time.Time, error) {
+	re := regexp.MustCompile(`(\d+)([a-zA-Z])`)
```

Comment:
We should probably be stricter with the regexp here:

`^(\d+)([my])$`

Actually, it might be simpler to do `!strings.HasPrefix(duration, "m") && !strings.HasPrefix(duration, "y")` to detect an invalid ending then call `Atoi(duration[:len(duration)-1])` to parse the number, switch'ing on `duration[len(duration)-1]` to create the correct time.

Reply:
Obviously, that should be `HasSuffix`, not `HasPrefix`.